### PR TITLE
[2.7][FrameworkBundle] Removed the use of TableHelper

### PR DESCRIPTION
**Pull request by saro0h**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

The PR https://github.com/symfony/symfony/pull/13121 on 3.0 removed the `TableHelper` class. Now the tests don't pass when using `components=high` version of dependencis, because of the use of the deprecated TableHelper.

This one removes the use of `TableHelper` without the removal of the `TableHelper` class, and adapt the existing code to use the Table class instead.

### CODE REVIEW

**Comment by saro0h** [comment]
Ping @nicolas-grekas : Can you take a look at that one?

The tests on components=high pass, but now, it's the ``components=low`` that has problems. Should I do the PR on 2.5 instead?

**Comment by fabpot** [comment]
@saro0h IMO, and when possible, we should remove usage of deprecated classes in the oldest possible branch.

**Comment by saro0h** [comment]
I definitely can but I'll be obliged to do 2 PR (or more) because of the evolution of code between version. 

Here is one exemple:
* 2.7: ``TextDescriptor::describeEventDispatcherListeners()`` exists and uses the TableHelper
* 2.5: The method does not exist at all.

Do you want me to do the work of removing the deprecated on all branches maintained on Symfony?

**Comment by xabbuh** [comment]
This is basically the same as #12970.

**Comment by saro0h** [comment]
@xabbuh No it's not: the code changed between the two branches.

**Comment by xabbuh** [comment]
@saro0h Indeed, I missed that change. Thanks.

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ protected function describeRouteCollection(RouteCollection $routes, array $optio
         }
 
         $this->writeText($this->formatSection('router', 'Current routes')."\n", $options);
-        $this->renderTable($table, !(isset($options['raw_output']) && $options['raw_output']));
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: You now don't make use of the decorated output anymore.

Comment (saro0h) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: Actually, in 3.0, I removed that also to make it more "standard".

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 2]
```diff
@@ -302,31 +300,29 @@ protected function describeEventDispatcherListeners(EventDispatcherInterface $ev
...
 
                 foreach ($eventListeners as $order => $eventListener) {
                     $table->addRow(array(sprintf('#%d', $order + 1), $this->formatCallable($eventListener)));
                 }
-
-                $this->renderTable($table);
             }
         }
+
+        $table->render();
```
Comment (hhamon) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: Moving this here means you will output a table with headers but without any content rows. It's not exactly the same behavior.

**Comment by hhamon** [comment]
Looks good!

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 3]
```diff
@@ -302,6 +302,8 @@ protected function describeEventDispatcherListeners(EventDispatcherInterface $ev
         $this->writeText($this->formatSection('event_dispatcher', $label)."\n", $options);
 
         $registeredListeners = $eventDispatcher->getListeners($event);
+        $table = new Table($this->getOutput());
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: looks like it's not needed

**Comment by fabpot** [comment]
Thank you @saro0h.

**Pull request merged**